src/host/pages/IMUMonitorPage.cpp

```diff
@@ -0,0 +1,102 @@
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+
+#include <stdio.h>
+#include "common/signalk/SKUpdate.h"
+#include "IMUMonitorPage.h"
+#include <KBoxLogging.h>
+#include "signalk/SKUnits.h"
+
+IMUMonitorPage::IMUMonitorPage(IMUConfig &config, SKHub& hub, IMUService &imuService) : _config(config), _imuService(imuService) {
+
+  static const int col1 = 5;
+  static const int col2 = 200;
+  static const int row1 = 26;
+  static const int row2 = 50;
+  static const int row3 = 152;
+  static const int row4 = 182;
+
+  addLayer(new TextLayer(Point(col1, row1), Size(20, 20), "HDG ° Mag", ColorWhite, ColorBlack, FontDefault));
+  addLayer(new TextLayer(Point(col2, row1), Size(20, 20), "Cal: Mag/Acc", ColorWhite, ColorBlack, FontDefault));
+  addLayer(new TextLayer(Point(col1, row3), Size(20, 20), "Heel °", ColorWhite, ColorBlack, FontDefault));
+  addLayer(new TextLayer(Point(col2, row3), Size(20, 20), "Pitch °", ColorWhite, ColorBlack, FontDefault));
+
+  _hdgTL = new TextLayer(Point(col1, row2), Size(20, 20), "--", ColorWhite, ColorBlack, FontLarge);
+  _calTL = new TextLayer(Point(col2, row2), Size(20, 20), "--", ColorWhite, ColorBlack, FontLarge);
+  _rollTL  = new TextLayer(Point(col1, row4), Size(20, 20), "--", ColorWhite, ColorBlack, FontLarge);
+  _pitchTL = new TextLayer(Point(col2, row4), Size(20, 20), "--", ColorWhite, ColorBlack, FontLarge);
+
+  addLayer(_hdgTL);
+  addLayer(_calTL);
+  addLayer(_rollTL);
+  addLayer(_pitchTL);
+
+  hub.subscribe(this);
+}
+
+bool IMUMonitorPage::processEvent(const ButtonEvent &be){
+  // DEBUG("EventTypeButton: %i", be.clickType);
+  if (be.clickType == ButtonEventTypeClick) {
+    DEBUG("Button ButtonEventTypeClick !!!");
+    // Change page on single click.
+    return false;
+  }
+  if (be.clickType == ButtonEventTypeLongClick) {
+    _imuService.setOffset();
+  }
+  return true;
+}
+
+bool IMUMonitorPage::processEvent(const TickEvent &te){
+  _imuService.getLastValues(_accelCalibration, _pitch, _roll, _magCalibration, _heading);
+  //DEBUG("AccelCalibration: %i | MagCalibration: %i", _accelCalibration, _magCalibration);
+
+  // TODO: Some damping for the display
+  _hdgTL->setText(String( SKRadToDeg(_heading), 1) + "°        ");
+  _calTL->setText(String( _magCalibration) + "/" + String( _accelCalibration) + "   ");
+  _pitchTL->setText(String( SKRadToDeg(_pitch), 1) + "°     ");
+  _rollTL->setText(String( SKRadToDeg(_roll), 1) + "°     ");
+
+  // Always show Hdg from IMU-sensor, but if the value is not trusted (which means
+  // calibration below config setting, change color to red
+	if ((_magCalibration <= _config.calHdg)||(_accelCalibration <= _config.calHeelPitch)) {
```

Comment:
nitpick: indentation problem here

Reply:
I think the IMUService should expose a `isMagCalibrated()` and is `isHeelAndPitchCalibrated()` methods. This way the actual technique we use to decide if we are calibrated is not duplicated in two completely different part of the code.

Reply:
I do not know how to get rid of ident-problem. In every editor I have it shows a 2-ident tab.
In Github it shows wrong. If I correct 8 -> 2 in github I can not save because it is said "no changes"

Reply:
Re-Indent: this article might help: https://stackoverflow.com/questions/4119872/netbeans-removing-trailing-whitespace-on-save-and-tabs-to-spaces

I think your editor is configured to show tabs as 2 spaces. this is why it looks good to you. But since they are actually tabs, it does not look good in github (they do this on purpose to make it easier to detect).

I can push a commit to fix it if needed.